packages/plugins/graphql/server/services/builders/type.js

```diff
@@ -322,13 +305,8 @@ module.exports = context => {
                */
               const options = { builder, attributeName, attribute, contentType, context };
 
-              // Enums
-              if (isEnumeration(attribute)) {
-                addEnumAttribute(options);
-              }
-
               // Scalars
-              else if (isStrapiScalar(attribute)) {
+              if (isStrapiScalar(attribute)) {
```

Comment:
Is there the specific reason you remove codes? It reverts the fix in #12005

### `master`
<img width="908" alt="before" src="https://user-images.githubusercontent.com/4951716/149791523-1addfc78-5451-46ae-8e62-1171911f5172.png">

### this PR
<img width="916" alt="after" src="https://user-images.githubusercontent.com/4951716/149791531-4d41246b-266f-4e32-a7ff-47ecc699401b.png">

Reply:
Good eye :)
The input stays as an enum but the output is put as a string so special characters could be return. I put an example in the PR description. Do you see a problem with that?

Reply:
I see, I just found that special characters are not supported in GraphQL specs..
http://spec.graphql.org/October2021/#note-46408
> Names in GraphQL are limited to the Latin ASCII subset of SourceCharacter in order to support interoperation with as many other systems as possible.

I think there will be more advantages of using the string type than using the enum type :)

Reply:
Alright, thank you :)

Reply:
Was just about to ask if this is a good idea since we just had an issue that GraphQL was returning strings for enums.... :thinking: It would be nice to put this issue to bed honestly

Reply:
What was the issue ?

Reply:
https://github.com/strapi/strapi/issues/12127

Reply:
https://github.com/strapi/strapi/issues/11993

Reply:
iicdii fixed this in https://github.com/strapi/strapi/pull/12005

Reply:
@iicdii considering that this PR only change the output schema and not the input, do you think it would create problem for the use cases of people listed above ? I'm not familliar with typescript. @Convly what do you think ?

Reply:
**It's trade-off.** It lose benefit of autogenerating api documentation If you force to enum to be a string. Otherwise, you can't use special characters because of GraphQL spec.
**The solution I thought of is to give the user an option to use special characters or not.** If user checked to use special chracters, enum types are shown as string. Otherwise, enum types are shown as enum, and the validation should be applied not to allow special characters.

Reply:
Thank you, it seems a reasonable proposition to put a config option for that. What do you think @alexandrebodin @Convly and @derrickmehaffy ?

Reply:
Makes sense to me, though I would advise by default we shouldn't allow it.

Reply:
Having enums types as output fields means that you know (thanks to the schema) what to expect from this attribute, which can be helpful. IMO we shouldn't accept special characters in the enum's name but should allow specifying different keys/values.
A config option can be handy and if it's really wanted, why not. However, I've to admit I find this a bit overkill as it will add edge-cases & complexity in the way we generate those enums.
So, again, depends if it's really necessary.